rust-runtime/aws-smithy-http-server/src/extension.rs

```diff
@@ -70,15 +72,15 @@ pub struct OperationExtension {
 #[derive(Debug, Clone, Error)]
 #[non_exhaustive]
 pub enum ParseError {
-    #[error("# was not found - missing namespace")]
+    #[error(". was not found - missing namespace")]
     MissingNamespace,
 }
 
 impl OperationExtension {
-    /// Creates a new [`OperationExtension`] from the absolute shape ID of the operation.
+    /// Creates a new [`OperationExtension`] from the absolute shape ID of the operation with `#` symbol replaced with a `.`.
     pub fn new(absolute_operation_id: &'static str) -> Result<Self, ParseError> {
         let (namespace, name) = absolute_operation_id
-            .split_once('#')
+            .rsplit_once('.')
```

Comment:
`rsplit` seems to work well,l though I worry that the caller accidentally pass a wrong shape ID, such as `com.amazonaws.ebs`. And then we interpret it as namespace `com.amazonaws` and name `ebs`.

It is unlikely to happen IMO but could we find a syntax less ambiguous?

Reply:
@crisidev Any good ideas?

Reply:
The only thing I can think about it? Is it possible to have a service that is dottet? as in com.amazon.ebs.My.Service? I don't know..

With the old code it would be com.amazon.ebs#My.Service